packages/integration-tests/src/EntityManager.reactiveRules.test.tsx

```diff
@@ -178,16 +178,41 @@ describe("EntityManager.reactiveRules", () => {
     expect(a.numberOfBooks.get).toBe(0);
   });
 
+  it.withCtx("updates derived fields that are foreign keys", async ({ em }) => {
+    // Given an entity with an async derived field
+    const a = newAuthor(em);
+    const b = newBook(em, { author: a, reviews: [{ rating: 10 }] });
+    await em.flush();
+
+    const favoriteBook = a.favoriteBook.get;
+    // For some reason .toMatchEntity(b) fails here
```

Comment:
I wasn't able to get `.toMatchEntity` to work, it will keep trying to match Collections to Arrays.